src/liballoc/boxed.rs

```diff
@@ -1070,3 +1071,10 @@ impl<F: ?Sized + Future + Unpin> Future for Box<F> {
         F::poll(Pin::new(&mut *self), cx)
     }
 }
+
+#[unstable(feature = "box_str_from_iter", issue = "0")]
```

Comment:
impls are going to be insta-stable, you can probably make this just be stable with the same feature name